hoodie-hive/src/main/java/com/uber/hoodie/hive/HiveSyncTool.java

```diff
@@ -19,64 +19,161 @@
-                        .jdbcPassword(cfg.hivePass)
-                        .build();
+  public void syncHoodieTable() {
+    LOG.info("Trying to sync hoodie table" + cfg.tableName + " with base path " + hoodieHiveClient
+        .getBasePath() + " of type " + hoodieHiveClient
+        .getTableType());
+    // Check if the necessary table exists
+    boolean tableExists = hoodieHiveClient.doesTableExist();
+    // Get the parquet schema for this dataset looking at the latest commit
+    MessageType schema = hoodieHiveClient.getDataSchema();
+    // Sync schema if needed
+    syncSchema(tableExists, schema);
 
-        HoodieDatasetReference datasetReference =
-                new HoodieDatasetReference(cfg.tableName, cfg.basePath, cfg.databaseName);
+    LOG.info("Schema sync complete. Syncing partitions for " + cfg.tableName);
+    // Get the last time we successfully synced partitions
+    Optional<String> lastCommitTimeSynced = Optional.empty();
+    if (tableExists) {
+      lastCommitTimeSynced = hoodieHiveClient.getLastCommitTimeSynced();
+    }
+    LOG.info("Last commit time synced was found to be " + lastCommitTimeSynced.orElse("null"));
+    List<String> writtenPartitionsSince = hoodieHiveClient
+        .getPartitionsWrittenToSince(lastCommitTimeSynced);
+    LOG.info("Storage partitions scan complete. Found " + writtenPartitionsSince.size());
+    // Sync the partitions if needed
+    syncPartitions(writtenPartitionsSince);
 
-        // initialize the strategies
-        PartitionStrategy partitionStrategy = new DayBasedPartitionStrategy();
-        SchemaStrategy schemaStrategy = new ParseSchemaFromDataStrategy();
+    hoodieHiveClient.updateLastCommitTimeSynced();
+    LOG.info("Sync complete for " + cfg.tableName);
 
-        // Creates a new dataset which reflects the state at the time of creation
-        HoodieHiveDatasetSyncTask datasetSyncTask =
-                HoodieHiveDatasetSyncTask.newBuilder().withReference(datasetReference)
-                        .withConfiguration(apiConfig).partitionStrategy(partitionStrategy)
-                        .schemaStrategy(schemaStrategy).build();
+    hoodieHiveClient.close();
+  }
 
-        // Sync dataset
-        datasetSyncTask.sync();
+  /**
+   * Get the latest schema from the last commit and check if its in sync with the hive table schema.
+   * If not, evolves the table schema.
+   *
+   * @param tableExists - does table exist
+   * @param schema - extracted schema
+   */
+  private void syncSchema(boolean tableExists, MessageType schema) {
+    // Check and sync schema
+    if (!tableExists) {
+      LOG.info("Table " + cfg.tableName + " is not found. Creating it");
+      switch (hoodieHiveClient.getTableType()) {
+        case COPY_ON_WRITE:
+          hoodieHiveClient.createTable(schema, HoodieInputFormat.class.getName(),
+              MapredParquetOutputFormat.class.getName(), ParquetHiveSerDe.class.getName());
+          break;
+        case MERGE_ON_READ:
```

Comment:
Yeah left it as a TODO comment for now. We need to figure out some details before we register the RO table. Mostly around major compaction (we only do minor IO bound compactions). Will take this up seperately